containers/ddev-webserver/ddev-webserver-base-scripts/start.sh

```diff
@@ -93,8 +93,28 @@ ls /var/www/html >/dev/null || (echo "/var/www/html does not seem to be healthy/
 # Make sure the TERMINUS_CACHE_DIR (/mnt/ddev-global-cache/terminus/cache) exists
 sudo mkdir -p ${TERMINUS_CACHE_DIR}
 
-sudo mkdir -p /mnt/ddev-global-cache/{bashhistory/${HOSTNAME},mysqlhistory/${HOSTNAME},nvm_dir/${HOSTNAME},npm,yarn/classic,yarn/berry,corepack}
+sudo mkdir -p /mnt/ddev-global-cache/{bashhistory/${HOSTNAME},mysqlhistory/${HOSTNAME},n_prefix/${HOSTNAME},nvm_dir/${HOSTNAME},npm,yarn/classic,yarn/berry,corepack}
 sudo chown -R "$(id -u):$(id -g)" /mnt/ddev-global-cache/ /var/lib/php
+
+if [ "${N_PREFIX:-}" != "" ] && [ "${N_INSTALL_VERSION:-}" != "" ]; then
+  ln -sf /mnt/ddev-global-cache/n_prefix/${HOSTNAME} "${N_PREFIX}"
+  # try a normal install that also uses cache and try again offline if it fails
+  n_install_result=true
+  timeout 30 n install "${N_INSTALL_VERSION}" 2> >(tee /tmp/n-install-stderr.txt >&2) || n_install_result=false
+  if [ "${n_install_result}" = "false" ]; then
+    timeout 30 n install "${N_INSTALL_VERSION}" --offline 2> >(tee -a /tmp/n-install-stderr.txt >&2) && n_install_result=true
+  fi
+  if [ "${n_install_result}" = "true" ]; then
+    ln -sf ${N_PREFIX}/bin/node ${N_PREFIX}/bin/nodejs
+    # we don't need this file if everything is fine
+    rm -f /tmp/n-install-stderr.txt
+  else
+    # remove the symlink on error so that the system Node.js can be used
+    rm -f ${N_PREFIX}
+  fi
+  hash -r
+fi
```

Comment:
Could we try moving this into a script that can be maintained separately? Like /usr/local/bin/n_install.sh or something?